### ISSUE TITLE
nextPld guards its preconditions with std.debug.assert, which is a no-op in ReleaseFast

### ISSUE THREAD

**Issue by JaySNL** [ddalcu/mlx-serve#97]
`Generator.nextPld` opens with:

```zig
std.debug.assert(self.sampling.constraint == null); // PLD + grammar not supported
std.debug.assert(self.logprobs_n == 0);             // PLD + logprobs not supported
```

`src/generate.zig:1619-1620`. `nextMtp` has the equivalent pair.

Release builds (`-Doptimize=ReleaseFast`, what `zig build` produces for the shipped binary) compile these out, so the invariants are documentation rather than enforcement in the build users actually run.

**This is latent today, not a live bug.** Every dispatch path currently pre-filters both conditions before a slot with `enable_pld` reaches the scheduler, so `nextPld` is never entered with a constraint or with logprobs requested. Nothing is broken as shipped.

Raising it because the failure mode is unusually bad for how easy the mistake is. If any future path lets a constrained request through — a new endpoint, a refactor of the mode-selection logic, a per-request override added somewhere — the result is not a crash or an error to the client. `nextPld` never reads `pending_logits` / `has_pending_logits`, so on the grammar path it treats the constrained branch's placeholder `next_token_id = 0` (`generate.zig:1235`) as a real committed token, forwards it at the wrong cache position, and continues: **silently off-schema output** returned as a normal 200. For the logprobs case the client silently gets no logprobs rather than an error.

I hit exactly this while carrying a local patch that widened which requests could reach PLD. In a debug build the assert fires immediately and the mistake is obvious; in ReleaseFast it produced plausible-looking JSON that did not match the requested schema, and took a code review to find rather than a test run.

### Suggestion

Make the invariant hold in release too — either an explicit early return / error on the release path, or `std.debug.assert` → a checked branch, at whichever of the two layers you'd rather own it:

- inside `nextPld` / `nextMtp` (defensive, cheap, once per request), or
- at the single dispatch site in the scheduler, if you'd rather keep the generators assuming their contract.

Happy to send a PR for whichever you prefer, or to leave it if you'd rather keep the invariant enforced purely by the callers.

### Environment

mlx-serve v26.7.9, macOS 26.2, M4 Max, `Qwen3.6-35B-A3B` 4-bit.

**Comment by ddalcu** [comment]
fixed in v26.7.10 thanks 👍
